### DISCUSSION TITLE
Custom Parameter related to streams

### DISCUSSION THREAD

**Discussion by Tabarane** [bluenviron/mediamtx#486] [Questions] [answered]
Hi, I was wondering if we can add additional custom parameter within the stream path ,

custom_value1 : xyzzy

paths:
  proxied:
    source: rtsp://original-url
    sourceOnDemand: yes
    CamCustomID: xyzzy
    StreamName: Xyzzy
    CamOwnName : John 

and retrieve this parameter using API, this will help in integration with other application

**Comment by aler9** [comment]
Hello, custom parameters at the moment are not availab,e but you can obtain the same result by creating a map into the script that is interacting with the API, and use it to pull every parameter you need. i.e. (node):

```js
async function downloadFromAPI() {
    const myExtraParams = {
        "proxied": {
            "camCustomID": "xyzzy",
            "streamName": "Xyzzy",
            "camOwnName": "John",
        }
    };

    const res1 = await fetch("http://localhost:9997/v1/paths/list");
    const res2 = await res1.json();

    for (const path in res2.items) {
        console.log("extra params for path " + path + ":", myExtraParams[path]);
    }
}
```

**Comment by WHTLY** [comment, marked as answer]
You can also create "dummy" streams and update the configuration file as many times as you want. Then just request the configuration through the API and read the parameters :D

[a comment by an automated account is not shown]
